src/Symfony/Component/Security/Http/Firewall/AbstractAuthenticationListener.php

```diff
@@ -66,27 +67,29 @@
-     * @param HttpUtils                              $httpUtils             An HttpUtilsInterface instance
-     * @param string                                 $providerKey
-     * @param AuthenticationSuccessHandlerInterface  $successHandler
-     * @param AuthenticationFailureHandlerInterface  $failureHandler
-     * @param array                                  $options               An array of options for the processing of a
-     *                                                                      successful, or failed authentication attempt
-     * @param LoggerInterface                        $logger                A LoggerInterface instance
-     * @param EventDispatcherInterface               $dispatcher            An EventDispatcherInterface instance
+     * @param SecurityContextInterface|TokenStorageInterface $tokenStorage          A SecurityContext or a TokenStorageInterface instance
+     * @param AuthenticationManagerInterface                 $authenticationManager An AuthenticationManagerInterface instance
+     * @param SessionAuthenticationStrategyInterface         $sessionStrategy
+     * @param HttpUtils                                      $httpUtils             An HttpUtilsInterface instance
+     * @param string                                         $providerKey
+     * @param AuthenticationSuccessHandlerInterface          $successHandler
+     * @param AuthenticationFailureHandlerInterface          $failureHandler
+     * @param array                                          $options               An array of options for the processing of a
+     *                                                                              successful, or failed authentication attempt
+     * @param LoggerInterface                                $logger                A LoggerInterface instance
+     * @param EventDispatcherInterface                       $dispatcher            An EventDispatcherInterface instance
      *
      * @throws \InvalidArgumentException
+     *
+     * Passing a SecurityContextInterface as a first argument was deprecated in 2.7 and will be removed in 3.0
      */
-    public function __construct(SecurityContextInterface $securityContext, AuthenticationManagerInterface $authenticationManager, SessionAuthenticationStrategyInterface $sessionStrategy, HttpUtils $httpUtils, $providerKey, AuthenticationSuccessHandlerInterface $successHandler, AuthenticationFailureHandlerInterface $failureHandler, array $options = array(), LoggerInterface $logger = null, EventDispatcherInterface $dispatcher = null)
+    public function __construct($tokenStorage, AuthenticationManagerInterface $authenticationManager, SessionAuthenticationStrategyInterface $sessionStrategy, HttpUtils $httpUtils, $providerKey, AuthenticationSuccessHandlerInterface $successHandler, AuthenticationFailureHandlerInterface $failureHandler, array $options = array(), LoggerInterface $logger = null, EventDispatcherInterface $dispatcher = null)
```

Comment:
I suggest forcing ``symfony/security-http`` to use ``symfony/security-core`` 2.6+ and typehinting the TokenStorageInterface